app/controllers/user_controller.rb

```diff
@@ -537,6 +537,17 @@ def work_out_post_redirect
     # Redirect to front page later if nothing else specified
     params[:r] = "/" if params[:r].nil? && params[:token].nil?
 
+    # retrieve the user's previous PostRedirect token if they did not click
```

Comment:
Discussed this in Slack – gist of it was that I can't really tell what the problem was that this is fixing. It _seems_ to make sense, but without examples of problems its hard to assess.

We decided to pull this commit out of the PR and make a new PR with it for proper discussion.